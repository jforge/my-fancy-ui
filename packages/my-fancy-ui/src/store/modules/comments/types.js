// @flow

export type CommentIdType = string;
export type CommentType = {
  id: number,
  email: string,
  name: string,
  body: string
};
export type CommentsByIdType = {
  [CommentIdType]: CommentType
};

export type StateType = {
  commentsById: CommentsByIdType
};
