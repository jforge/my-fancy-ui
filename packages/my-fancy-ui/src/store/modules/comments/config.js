// @flow

export const namepsace = '@immowelt/my-fancy-ui';
export const moduleId = 'comments';
