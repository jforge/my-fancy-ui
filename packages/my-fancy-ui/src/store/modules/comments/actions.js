// @flow

import type {CommentType} from './types.js';

import {createAction} from 'redux-actions';
import {createActionTypeFactory} from 'redux-lumbergh';
import {moduleId, namepsace} from './config.js';

//
// Action types
//
const createActionType = createActionTypeFactory(namepsace, moduleId);

const actionTypes = {
  LOAD_COMMENTS: createActionType('LOAD_COMMENTS'),
  SET_COMMENTS: createActionType('SET_COMMENTS')
};

//
// Action creators
//
const loadComments = createAction(actionTypes.LOAD_COMMENTS);

export type SetCommentsPayloadType = {comments: Array<CommentType>};
const setComments = createAction(
  actionTypes.SET_COMMENTS,
  (comments: Array<CommentType> = []): SetCommentsPayloadType => ({
    comments
  })
);

const actions = {
  loadComments,
  setComments
};

export {actionTypes, actions};
