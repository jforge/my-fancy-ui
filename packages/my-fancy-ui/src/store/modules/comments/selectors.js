// @flow

import type {StateType} from './../../types.js';
import type {CommentType, CommentIdType, CommentsByIdType} from './types.js';

import {$get} from 'plow-js';
import {moduleId} from './config.js';

export const getCommentsById = (state: StateType): CommentsByIdType =>
  $get([moduleId, 'commentsById'], state);
export const getCommentForId = (
  state: StateType,
  props: {id: CommentIdType}
): CommentType => $get([moduleId, 'commentsById', props.id], state);
