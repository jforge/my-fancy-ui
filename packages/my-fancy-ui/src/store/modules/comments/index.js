// @flow

import type {CommentsByIdType, StateType} from './types.js';
import type {SetCommentsPayloadType} from './actions.js';

import {$set} from 'plow-js';
import {createReducer} from 'redux-lumbergh';
import sagas from './sagas/';
import * as memoizedSelectors from './selectors.memoized.js';
import * as inputSelectors from './selectors.js';
import {actions, actionTypes} from './actions.js';
import {moduleId} from './config.js';

//
// The state structure
//
const initialState: StateType = {
  commentsById: {}
};

//
// Action Handlers
//
const ACTION_HANDLERS = {
  [actionTypes.SET_COMMENTS]: (payload: SetCommentsPayloadType) => {
    const commentsById: CommentsByIdType = payload.comments.reduce(
      (byId, comment) => {
        byId[comment.id] = comment;

        return byId;
      },
      {}
    );

    return $set('commentsById', commentsById);
  }
};

//
// Reducer
//
const reducer = createReducer(initialState, ACTION_HANDLERS);

//
// Export all selectors as one object.
//
export const selectors = {
  ...memoizedSelectors,
  ...inputSelectors
};

export {moduleId, sagas, actions, actionTypes, initialState, reducer};
