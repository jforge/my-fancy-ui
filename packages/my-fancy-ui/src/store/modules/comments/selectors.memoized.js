// @flow

import type {CommentsByIdType, CommentIdType} from './types.js';

import {createSelector} from 'reselect';
import {getCommentsById} from './selectors.js';

export const getCommentIds = createSelector(
  [getCommentsById],
  (commentsById: CommentsByIdType): Array<CommentIdType> => {
    return Object.keys(commentsById);
  }
);
