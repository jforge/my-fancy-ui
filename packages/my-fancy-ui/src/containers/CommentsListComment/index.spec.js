import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {createMapStateToPropsSnapshot} from 'jest-react-redux';
import {selectors} from './../../store/modules/comments/';
import {CommentsListComment, mapStateToProps} from './index.js';

describe('<CommentsListComment/>', () => {
  let props;

  beforeEach(() => {
    props = {
      commentId: 'fooId',
      deleteComment: jest.fn(),
      comment: {
        id: 1,
        email: 'foo@bar.com',
        name: 'Foo Bar',
        body: 'Lorem ipsum'
      }
    };
  });

  it('should render correctly.', () => {
    const wrapper = shallow(<CommentsListComment {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('mapStateToProps()', () => {
  it('should be a function', () => {
    expect(typeof mapStateToProps).toBe('function');
  });

  it('should execute all given selectors with the state and return an object containing the state props.', () => {
    const result = createMapStateToPropsSnapshot({
      mapStateToProps,
      selectors,
      selectorImplementationByKey: {
        getCommentForId: () => ({commentId: 'fooId'})
      },
      ownProps: {commentId: 'fooId'}
    });

    expect(result).toMatchSnapshot();
  });
});
