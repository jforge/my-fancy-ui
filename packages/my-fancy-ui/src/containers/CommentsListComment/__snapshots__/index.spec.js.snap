// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentsListComment/> should render correctly. 1`] = `
<div
  className="mdl-card mdl-card__supporting-text mdl-shadow--2dp"
  style={
    Object {
      "marginBottom": "1.5rem",
      "width": "100%",
    }
  }
>
  <div>
    <b>
      fooId
    </b>
    Â 
    Foo Bar
     (
    foo@bar.com
    )
  </div>
  <p
    className="comments__comment__body"
  >
    Lorem ipsum
  </p>
</div>
`;

exports[`mapStateToProps() should execute all given selectors with the state and return an object containing the state props. 1`] = `
Object {
  "calls": Object {
    "getCommentForId": Object {
      "argsByCallCount": Object {
        "0": Array [
          Object {
            "__mockState": true,
          },
          Object {
            "id": "fooId",
          },
        ],
      },
      "callCount": 1,
    },
  },
  "stateProps": Object {
    "comment": Object {
      "commentId": "fooId",
    },
  },
}
`;
